### ISSUE TITLE
Task: Convert Wiki to the new CommonMark

### ISSUE THREAD

**Issue by Griatch** [evennia/evennia#1260]
#### Brief summary of issue

Many Evennia Wiki links currently does not render correctly, rendering the raw text rather than producing a HTML link. The reason for this is that GitHub a few days ago switched to use CommonMark, a stricter form of MarkDown. Here is the reply I got from github support: 

> Thanks for reaching out! We recently updated our markdown renderer to CommonMark: https://github.com/blog/2333-a-formal-spec-for-github-flavored-markdown
> 
> The issue you're seeing is related to this update. CommonMark does not allow for space characters within URLs, so you'll need to either encode those as %20 or as - since that's what we turn it into in the URL for those pages.
>
> You can review the full spec for CommonMark and GitHub's extensions here: https://github.github.com/gfm/

#### Extra information

So far the main problem we have seen with this is the lack of working links since we use spaces in the reference in many places. We should instead use `-` to delimintate multiple words, so a link should be converted like this: 

    [Getting started](Getting Started) -> [Getting started](Getting-Started)

Note that Evennia wiki pages have invisible links at their tops, on the form 

    `[](Explains how to download, install and start Evennia for the first time.)`

These may look strange *but must remain*, and should be converted in the same way. They are used by Evennia's wiki indexing tool and is what is producing the [Wiki Index](https://github.com/evennia/evennia/wiki/Wiki-Index).

#### Implementation

This is a critically important thing to fix. All help is appreciated to convert where needed in the interim for critical pages, but with some 200 wiki pages, someone (most likely me) will need to make a parser to convert all these link texts automatically as soon as possible.

**Comment by Griatch** [comment]
@BlauFeuer Excellent, thanks. This should help until I have the time to write a parser to operate on the entire wiki repo.

**Comment by BlauFeuer** [comment]
I'm back editing again - after receiving the following kind reply: 
> Sorry about that! Sometimes our spam-detecting robots get a little overzealous and need to be reprogrammed. I've reset your profile, so your repositories and gists should be available again, and you shouldn't see that message anymore. 

I seeing this markup visible on some pages:  `<wiki:toc max_depth="3" />`
I've left it alone, but is it something that needs converting to new markup, also?
I haven't read the CommonMark spec yet, but I will later today.

**Comment by Griatch** [comment]
No, the `<wiki:...> ` bits are ancient leftovers from our Google Code days.  They serve no function anymore.

**Comment by BlauFeuer** [comment]
I've completed my run through the pages in alphabetical order from the pages list, but there are still some dead links in https://github.com/evennia/evennia/wiki/Wiki-Index and I did not edit that page since it is said to be auto-generated.  I fixed a few stray things along the way.  When I see a `<wiki:` tag, I'll remove it, but I won't go hunting for them.

Some pages did not have a wiki-index link at the top.  I added a few, but focused more on fixing all broken links I could find in non-auto-generated pages.  It's about as usable as it was before the markup engine change, but there's always work to do on the wiki.  If there's anything specific to focus on, just mention it as an issue and I'll do what I can to contribute.

**Comment by Griatch** [comment]
@BlauFeuer Thanks a lot for your work on this! I have no run the wiki through a converter to auto-convert all links that may remain. I also updated the wiki-index generator, api generator and default command page generators. I don't think I missed anything but well, time will tell.

I have over time removed the wiki indices when I've noticed them them but since they are not rendered they are not a high priority.

Closing as resolved.
